xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/macros.vm

```diff
@@ -65,6 +65,21 @@
 #macro(submitButton $name $shortcut $value $class)
 <input class="btn $!class" type="submit" name="$name"#if($keyboardShortcutsEnabled) title="$shortcut"#end value="$value"/>
 #end
+#**
+ * Returns the escaped value of the shortcut. This value can be set in two ways: via user preferences and wiki level translation
+ * keys. If there are both, the user preference will take priority.
+ * @param userPreferenceName The id of the user preference that is listed in XWikiUsersDocumentInitializer.java
+ *        e.g. `shortcut_developer_usertype`
+ * @param translationKey  The key of the translation containing the wiki level defined shortcut.
+ *        e.g. `core.shortcuts.developer.user.type`
+ * @since 16.9.0-rc-1
+ *#
+#macro(getShortcutValue $translationKey)
+#set ($shortcutPreferenceObject = $xwiki.getDocument($xcontext.getUser()).getObject('XWiki.UserPreferenceShortcutsClass', 'translationKey', $translationKey))
+#if("$!shortcutPreferenceObject"!='')$escapetool.javascript($escapetool.xml($shortcutPreferenceObject.value))##
+#{else}$escapetool.javascript($escapetool.xml($services.localization.render($translationKey)))##
```

Comment:
Escaping should be done by the caller, in particular if the value is used in different contexts.

Reply:
Updated this for https://github.com/xwiki/xwiki-platform/pull/2981#discussion_r1783076538 in https://github.com/xwiki/xwiki-platform/pull/2981/commits/74f90d2e6fb8da0fee19b1d1256e637f135e9b87